### PR TITLE
Upgrade to new travis infrastructure

### PR DESCRIPTION
**Pull request by daleharvey**

### CODE REVIEW

**Comment by daleharvey** [comment]
So Couch Master is working with this and even coverage seems to have been fixed, the tests still arent as stable as they seemed to have been before so going to carry on, but yay

**Comment by daleharvey** [comment]
```
$ sudo service couchdb start
couchdb: unrecognized service
$ sudo service docker start
start: Job is already running: docker
```

No idea what is up with travis with this

**Comment by klaemo** [comment]
You can also use docker to run the stable version of couchdb: `docker run -d -p 5984:5984 klaemo/couchdb:1.6.1`.

That allows you to easily run multiple couch versions side by side.

File: .travis.yml [hunk 1]
```diff
@@ -5,42 +5,30 @@ node_js:
...
-      - git
-      - libicu-dev
-      - libmozjs185-dev
-      - python
-      - g++
-      - g++-4.8
+
+before_install:
+  - docker pull klaemo/couchdb:2.0-dev
+  - docker run -d -p 15984:5984 klaemo/couchdb:2.0-dev --with-haproxy --with-admin-party-please -n 1
```
Comment (nickcolley) [inline .travis.yml]: https://github.com/pouchdb/pouchdb/pull/4666#discussion_r48094019

**Comment by daleharvey** [comment]
Yeh lets give that a shot, makes us less fragile to travis changes

File: .travis.yml [hunk 2]
```diff
@@ -4,43 +4,37 @@ node_js:
...
-      - libmozjs185-dev
-      - python
-      - g++
-      - g++-4.8
+
+before_install:
+  # Install CouchDB Stable
+  - docker run -d -p 5984:5984 klaemo/couchdb:1.6.1
+  - docker pull klaemo/couchdb:1.6.1
+
```
Comment (klaemo) [inline .travis.yml]: wrong order. also `docker run` pulls the image implicitly.

File: .travis.yml [hunk 3]
```diff
@@ -4,43 +4,35 @@ node_js:
...
-      - python
-      - g++
-      - g++-4.8
+
+before_install:
+  # Install CouchDB Stable
+  - docker run -d -p 5984:5984 klaemo/couchdb:1.6.1
+
+  # Install CouchDB Master
+  - docker run -d -p 15984:5984 klaemo/couchdb:2.0-dev --with-haproxy --with-admin-party-please -n 1
```
Comment (nolanlawson) [inline .travis.yml]: this is fantastic

**Comment by nolanlawson** [comment]
So much red! I love it. +1 when green.

**Comment by nolanlawson** [comment]
Hm, I see three failed tests, and they all appear to be due to timeouts:

     Error: timeout of 50000ms exceeded. Ensure the done() callback is being called in this test.

(Different tests every time.)

I'm worried that this new Travis infrastructure is slower than the last one, so we're going to need to increase timeouts. But in general I am still +1. I'll restart the tests.

**Comment by klaemo** [comment]
Are you sure it's running on the new infrastructure? It is supposed to be faster...

The travis docs say you enable the new infra with `sudo: false`, but in order to use docker they want you to `sudo: required`. Confusing.

**Comment by daleharvey** [comment]
This will go green with enough restarts (yay)

After this, we may have a leveldown issue still, we certainly have a phantomjs issue, its pretty much permafail right now and then I think we will be somewhat back to the same level of stability we were last month (~1/15 failures)

**Comment by nolanlawson** [comment]
Seeing some `/home/travis/build.sh: line 45: docker: command not found`.

**Comment by daleharvey** [comment]
ok those should be fixed now, fingers crossed

**Comment by daleharvey** [comment]
ok, given the current state of the test suite this is green, following up with the phantom errors - https://github.com/pouchdb/pouchdb/commit/203044c9824d8eb444640df12a8ea1b69260103e

**Pull request closed without merging**